src/client/eppo-client.spec.ts

```diff
@@ -596,13 +600,223 @@ describe('EppoClient E2E test', () => {
 
     const client = new EppoClient(storage);
     let assignment = client.getAssignment('subject-10', flagKey, { appVersion: 9 });
-    expect(assignment).toEqual(null);
+    expect(assignment).toBeNull();
     assignment = client.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual(null);
+    expect(assignment).toBeNull();
     assignment = client.getAssignment('subject-10', flagKey, { appVersion: 11 });
     expect(assignment).toEqual('control');
   });
 
+  it('returns control variation and logs status_quo if subject is in holdout', () => {
+    const entry = {
+      ...mockExperimentConfig,
+      allocations: {
+        allocation1: {
+          percentExposure: 1,
+          holdout: {
+            statusQuo: 'variation-7',
+            shipped: null,
+            percentExposure: 0.2,
+            keys: [
+              {
+                holdoutKey: 'holdout-2',
+                statusQuoShardRange: {
+                  start: 0,
+                  end: 50,
+                },
+              },
+              {
+                holdoutKey: 'holdout-3',
+                statusQuoShardRange: {
+                  start: 51,
+                  end: 100,
+                },
+              },
+            ],
+          },
+          variations: [
+            {
+              name: 'control',
+              value: 'control',
+              typedValue: 'control',
+              shardRange: {
+                start: 0,
+                end: 34,
+              },
+              variationKey: 'variation-7',
+            },
+            {
+              name: 'variant-1',
+              value: 'variant-1',
+              typedValue: 'variant-1',
+              shardRange: {
+                start: 34,
+                end: 67,
+              },
+              variationKey: 'variation-8',
+            },
+            {
+              name: 'variant-2',
+              value: 'variant-2',
+              typedValue: 'variant-2',
+              shardRange: {
+                start: 67,
+                end: 100,
```

Comment:
The hard-coded shard space is 10,000 (0-9999) so I'd just have the tests mirror that.